### PR TITLE
feat: Apply filter on JavaOutputProcessor to print only needed classes.

### PR DESCRIPTION
**Pull request by GerardPaligot**

### CODE REVIEW

File: src/main/java/spoon/Launcher.java [hunk 1]
```diff
@@ -371,6 +376,15 @@ protected static JSAP defineArgs() {
 			sw1.setDefault("false");
 			jsap.registerParameter(sw1);
 
+			// Generate only java files specified.
+			opt2 = new FlaggedOption("generate-files");
+			opt2.setShortFlag('f');
+			opt2.setLongFlag("generate-files");
+			opt2.setHelp("Generate only java files specified by their fully qualified name and separated by ':'. This argument only works when your output-type is specified at classes.");
```
Comment (monperrus) [inline src/main/java/spoon/Launcher.java]: Only generate the given fully qualified java classes  (separated by ':' if multiple are given)

File: src/main/java/spoon/SpoonAPI.java [hunk 2]
```diff
@@ -56,6 +58,22 @@
...
+	 */
+	void setFilteringOutputSource(Filter<CtType<?>> typeFilter);
+
+	/**
+	 * Creates a filter from names in argument when the output-type is defined on classes.
+	 *
+	 * @param names
+	 * 		Names of class in fully qualified name to know which type Spoon must print.
+	 */
+	void setFilteringOutputSource(String... names);
```
Comment (monperrus) [inline src/main/java/spoon/SpoonAPI.java]: qualifedNames

File: src/main/java/spoon/SpoonAPI.java [hunk 3]
```diff
@@ -56,6 +58,22 @@
 	void setSourceOutputDirectory(File outputDirectory);
 
 	/**
+	 * Applies a filter when the output-type is defined on classes.
+	 *
+	 * @param typeFilter
+	 * 		Filter on CtType to know which type Spoon must print.
+	 */
+	void setFilteringOutputSource(Filter<CtType<?>> typeFilter);
```
Comment (monperrus) [inline src/main/java/spoon/SpoonAPI.java]: setOutputFilter

**Comment by GerardPaligot** [comment]
@monperrus PR ok.

**Pull request merged**